### ISSUE TITLE
fromSRA - illegal character error

### ISSUE THREAD

**Issue by sstadick** [nextflow-io/nextflow#1700]
## Bug report 

(Please follow this template replacing the text between parentheses with the requested information)

### Expected behavior and actual behavior

I expect the `.fromSRA` method to pull fastqs as specified in the docs. I get an `Illegal character in authority at index...` error.

### Steps to reproduce the problem

Running example from docs: https://www.nextflow.io/docs/latest/channel.html?highlight=sra#fromsra

```nextflow
ids = ['ERR908507', 'ERR908506', 'ERR908505']
Channel.fromSRA(ids).view()
```

### Program output 

```
N E X T F L O W  ~  version 20.07.1
Launching `srapuller.nf` [tender_joliot] - revision: 92f3e673bf
WARN: Define the NCBI_API_KEY env variable to use NCBI search service -- Read more https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/
Illegal character in authority at index 6: ftp://err908507      ftp.sra.ebi.ac.uk
```
.nextflow.log
```
Aug-06 12:05:43.689 [main] DEBUG nextflow.cli.Launcher - $> nextflow srapuller.nf
Aug-06 12:05:43.808 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 20.07.1
Aug-06 12:05:43.816 [main] INFO  nextflow.cli.CmdRun - Launching `srapuller.nf` [tender_joliot] - revision: 92f3e673bf
Aug-06 12:05:43.887 [main] DEBUG nextflow.Session - Session uuid: 80652a1d-8f24-456a-babc-a3276e8d816a
Aug-06 12:05:43.887 [main] DEBUG nextflow.Session - Run name: tender_joliot
Aug-06 12:05:43.888 [main] DEBUG nextflow.Session - Executor pool size: 16
Aug-06 12:05:53.909 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 20.07.1 build 5412
  Created: 24-07-2020 15:18 UTC (09:18 MDT)
  System: Mac OS X 10.15.3
  Runtime: Groovy 2.5.11 on OpenJDK 64-Bit Server VM GraalVM CE 20.0.0 1.8.0_242-b06
  Encoding: UTF-8 (UTF-8)
  Process: 21122@Seths-MacBook-Pro.local [10.0.0.63]
  CPUs: 16 - Mem: 32 GB (1.2 GB) - Swap: 16 GB (1.5 GB)
Aug-06 12:05:53.923 [main] DEBUG nextflow.Session - Work-dir: /Users/sstadick/dev/misc/sratest/work [Mac OS X]
Aug-06 12:05:53.924 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /Users/sstadick/dev/misc/sratest/bin
Aug-06 12:05:53.976 [main] DEBUG nextflow.Session - Observer factory: TowerFactory
Aug-06 12:05:53.978 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Aug-06 12:05:54.119 [main] DEBUG nextflow.Session - Session start invoked
Aug-06 12:05:54.474 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Aug-06 12:05:54.494 [main] DEBUG nextflow.Session - Workflow process names [dsl1]: 
Aug-06 12:05:54.522 [ForkJoinPool.commonPool-worker-9] WARN  nextflow.datasource.SraExplorer - Define the NCBI_API_KEY env variable to use NCBI search service -- Read more https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/
Aug-06 12:05:54.529 [ForkJoinPool.commonPool-worker-9] DEBUG nextflow.datasource.SraExplorer - SRA search url=https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=sra&usehistory=y&retmode=json&term=ERR908507
Aug-06 12:05:54.580 [main] DEBUG nextflow.script.ScriptRunner - > Await termination 
Aug-06 12:05:54.580 [main] DEBUG nextflow.Session - Session await
Aug-06 12:05:54.580 [main] DEBUG nextflow.Session - Session await > all process finished
Aug-06 12:05:54.580 [main] DEBUG nextflow.Session - Session await > all barriers passed
Aug-06 12:05:55.035 [ForkJoinPool.commonPool-worker-9] DEBUG nextflow.datasource.SraExplorer - SRA data request url=https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=sra&retmode=json&query_key=1&WebEnv=NCID_1_55387103_130.14.22.33_9001_1596737154_323263209_0MetA0_S_MegaStore&retstart=0&retmax=1000
Aug-06 12:05:55.161 [ForkJoinPool.commonPool-worker-9] ERROR nextflow.Channel - Illegal character in authority at index 6: ftp://err908507	ftp.sra.ebi.ac.uk
java.net.URISyntaxException: Illegal character in authority at index 6: ftp://err908507	ftp.sra.ebi.ac.uk
	at java.net.URI$Parser.fail(URI.java:2848)
	at java.net.URI$Parser.parseAuthority(URI.java:3186)
	at java.net.URI$Parser.parseHierarchical(URI.java:3097)
	at java.net.URI$Parser.parse(URI.java:3053)
	at java.net.URI.<init>(URI.java:588)
	at nextflow.file.http.XFileSystemProvider.key(XFileSystemProvider.groovy:62)
	at nextflow.file.http.XFileSystemProvider.key(XFileSystemProvider.groovy:66)
	at nextflow.file.http.XFileSystemProvider.getFileSystem(XFileSystemProvider.groovy:123)
	at nextflow.file.http.XFileSystemProvider.getPath(XFileSystemProvider.groovy:138)
	at java.nio.file.Paths.get(Paths.java:143)
	at nextflow.file.FileHelper.asPath(FileHelper.groovy:278)
	at nextflow.file.FileHelper.asPath(FileHelper.groovy:261)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:101)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite$StaticMetaMethodSiteNoUnwrap.invoke(StaticMetaMethodSite.java:131)
	at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite.call(StaticMetaMethodSite.java:89)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at nextflow.datasource.SraExplorer.getFastqUrl(SraExplorer.groovy:281)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:190)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:168)
	at nextflow.datasource.SraExplorer.parseDataResponse(SraExplorer.groovy:200)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:190)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:168)
	at nextflow.datasource.SraExplorer.query1(SraExplorer.groovy:157)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:190)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:168)
	at nextflow.datasource.SraExplorer.query0(SraExplorer.groovy:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:190)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:168)
	at nextflow.datasource.SraExplorer.query0(SraExplorer.groovy:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:190)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:58)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:156)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:168)
	at nextflow.datasource.SraExplorer.apply(SraExplorer.groovy:103)
	at nextflow.Channel$_fetchSraFiles0_closure16.doCall(Channel.groovy:608)
	at nextflow.Channel$_fetchSraFiles0_closure16.call(Channel.groovy)
	at groovy.lang.Closure.run(Closure.java:486)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1632)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Aug-06 12:05:55.165 [ForkJoinPool.commonPool-worker-9] DEBUG nextflow.Session - Session aborted -- Cause: Illegal character in authority at index 6: ftp://err908507	ftp.sra.ebi.ac.uk
Aug-06 12:05:55.180 [ForkJoinPool.commonPool-worker-9] DEBUG nextflow.Session - The following nodes are still active:
  [operator] view

Aug-06 12:05:55.188 [main] DEBUG nextflow.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=0; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=0ms; failedDuration=0ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=0; peakCpus=0; peakMemory=0; ]
Aug-06 12:05:55.380 [main] DEBUG nextflow.CacheDB - Closing CacheDB done
Aug-06 12:05:55.384 [main] DEBUG nextflow.util.SpuriousDeps - AWS S3 uploader shutdown
Aug-06 12:05:55.398 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye
```
### Environment 

* Nextflow version: 20.07.1.5412
* Java version: 11.0.1 2018-10-16 LT
* Operating system: macOS
* Bash version: fish, version 3.1.0

### Additional Info

I did try with an apiKey as well, but that didn't fix the issue.

**Comment by sstadick** [comment]
See PR here: https://github.com/nextflow-io/nextflow/pull/1711

**Comment by pditommaso** [comment]
Solved by a45decfb.
